website/src/blogs/2017-08-31-ara-recording-ansible-playbook-runs.md

```diff
@@ -0,0 +1,44 @@
+---
+title: ARA - Recording Ansible Playbook Runs
+author: Karthik Satchitanand
+author_info: Karthik has been into the Design and Development of tools for infrastructure as code, software testing performance & benchmarking & chaos engineering.
+tags: Ansible, Automation, DevOps, Openebs
+date: 31-08-2017
+excerpt: Ansible playbooks can generate quite a lot of console data. Add the -v (I do it by default !) and you have a quite a lot to scroll through.
+---
+
+Ansible playbooks can generate quite a lot of console data. Add the -v (I do it by default !) and you have quite a lot to scroll through. Oftentimes, one tends to feel the need for a better reporting mechanism — one which is easy to scour for specific task status, whilst having the luxury to extract additional debug info for the same if needed. Also, it would be great if this aid is available for playbook runs across time, i.e., for older playbook runs as well. This would be especially beneficial when running ansible-based CI suites, like @OpenEBS, where the application deployments, test setup & execution are driven by Ansible.
+
+Sounds like a perfect requirement for a UI-based solution, doesn’t it?
+
+A lot of people using Ansible for medium-large scale deployments are known to use **Tower**, **Rundeck** or the opensource alternative **semaphore** to achieve this (Tower has a lifelong self-support trial license without a few features for under 10 nodes). There is also **Foreman**, which is great in environments where a hybrid tool set — a puppet with ansible, or chef with ansible, etc., is used. Most of these are workflow management tools that do more than *just* what we desired above, i.e, playbook recording. These tools provide a centralized management capability wherein inventory, users, task/play scheduling, notifications can all be controlled from a dashboard. Now, it is also possible to integrate these with other popular CI tools like Jenkins (watch [this](https://www.youtube.com/watch?v=CqjeIiHvy30&amp;feature=youtu.be)).
+
+However, if there is already a system in place to perform some of the things these tools do (dynamic inventories, playbook triggers/scheduling) while not really needing others (access control, graphs), and you are only looking for the ability to store and analyze playbook runs via UI, then **ARA** (Ansible Run Analysis) is your tool.
+
+![ARA (Ansible Run Analysis)](/images/blog/ansible-run-analysis.png)
+
+Heavily used by the OpenStack community in their CI projects, ARA is built to just *"record"* playbook runs (Read ARA’s [manifesto](http://ara.readthedocs.io/en/latest/manifesto.html#manifesto), to understand more about its narrow focus). It does this via an ansible callback plugin to store run details into a database and a web interface to visualize the database.
+
+![Viewing playbook tasks summary in ARA](/images/blog/playbook-tasks-summary-in-ara.png)  
+**Viewing playbook tasks summary in ARA!**
```

Comment:
The texts look like figure caption, please check if it's coming okay below the image or as a heading to the image